Tasks/DownloadBuildArtifact/main.ts

```diff
@@ -0,0 +1,48 @@
+import {BuildArtifact, ArtifactResource} from 'vso-node-api/interfaces/BuildInterfaces';
+import {WebApi, getBearerHandler} from 'vso-node-api/WebApi';
+import * as tl from 'vsts-task-lib/task';
+
+import {ArtifactProvider} from './ArtifactProvider';
+import {FileContainerProvider} from './FileContainer';
+
+async function main(): Promise<void> {
+	let projectId = tl.getVariable('System.TeamProjectId');
+	let buildId = parseInt(tl.getInput("buildId"));
+	let artifactName = tl.getInput("artifactName");
+	let downloadPath = tl.getPathInput("downloadPath");
+
+	let accessToken = getAuthToken();
+	let credentialHandler = getBearerHandler(accessToken);
+	let collectionUrl = tl.getVariable("System.TeamFoundationCollectionUri");
+	let vssConnection = new WebApi(collectionUrl, credentialHandler);
+
+	// get the artifact metadata
+	let buildApi = vssConnection.getBuildApi();
+	let artifact = await buildApi.getArtifact(buildId, artifactName, projectId);
```

Comment:
nice, it looks like we will be able to plug in fileshare easily.

i'm curious what your thoughts are about setting variables. can discuss offline.